src/wp-admin/includes/class-pclzip.php

```diff
@@ -4209,12 +4209,20 @@ function privExtractFileAsString(&$p_entry, &$p_string, &$p_options)
         if ($p_entry['compression'] == 0) {
 
           // ----- Reading the file
-          $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          if ($p_entry['compressed_size'] > 0) {
+            $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          } else {
+            $p_string = false;
```

Comment:
This adds a different and unexpected new type for this variable (boolean, while string is expected), while this is a variable being changed by reference.

Suggestion:
```php
            $p_string = '';
```

Reply:
`fread` returns `false` on failure (https://www.php.net/manual/en/function.fread.php), so this code doesn't change anything except avoiding fatal error on PHP 8.

Reply:
But I agree with you that an empty string looks better. I'll check that boolean values don't process especially and change `false` to `''`.